waiter/src/waiter/metrics_consumer.clj

```diff
@@ -0,0 +1,222 @@
+                                          (assoc watches-state metrics-service-url (query-state-fn include-flags)))
+                                        {}
+                                        metrics-service-watches)))))
+          go-chan
+          (async/go
+            (try
+              (loop [current-state @state-atom]
+                (reset! state-atom current-state)
+                (let [[msg current-chan]
+                      (async/alts! [exit-chan token-metric-chan query-chan])
+                      next-state
+                      (condp = current-chan
+                        exit-chan
+                        (do
+                          (log/warn "Stopping metrics-consumer-maintainer")
+                          (when (not= :exit msg)
+                            (throw (ex-info "Stopping metrics-consumer-maintainer" {:time (clock) :reason msg}))))
+
+                        token-metric-chan
+                        (timers/start-stop-time!
+                          (metrics/waiter-timer "core" "metrics-consumer" "handle-token-metric")
+                          (let [{:strs [descriptor lastRequestTime latest-descriptor token]} msg
+                                {fallback-service-id :service-id} descriptor
+                                {latest-service-id :service-id} latest-descriptor
+                                stored-last-request-time (get-in (service-id->metrics-fn) [fallback-service-id "last-request-time"])
+                                is-new-last-request-time? (or (nil? stored-last-request-time)
+                                                              (t/before? stored-last-request-time lastRequestTime))]
+                            (log/info "metrics consumer event" {:token token
+                                                                :stored-last-request-time stored-last-request-time
+                                                                :candidate-last-request-time lastRequestTime
+                                                                :fallback-service-id fallback-service-id
+                                                                :latest-service-id latest-service-id})
+                            (when
+                              (and is-new-last-request-time?
+                                   (leader?-fn))
+                              (timers/start-stop-time!
+                                (metrics/waiter-timer "core" "metrics-consumer" "update-last-request-time")
+                                (log/info "updating last request time for service" {:fallback-service-id fallback-service-id
+                                                                                    :last-request-time lastRequestTime})
+                                (send local-usage-agent metrics/update-last-request-time-usage-metric fallback-service-id lastRequestTime)
```

Comment:
Does not feel like local metrics updates should only be triggered on the leader. Any router connecting to the metrics service should be able to update its local metrics.